### ISSUE TITLE
Extension no longer working, option completely missing from UI

### ISSUE THREAD

**Issue by AlexADittrich** [intOrfloat/spotitySpeedExtension#7]
Recently updated Chrome, extension no longer working. The playback speed adjustment is now missing

![image](https://user-images.githubusercontent.com/19375198/111795843-910d3a80-8895-11eb-912c-6f0814633ca0.png)

Chrome:  Version 89.0.4389.90 (Official Build) (64-bit)

Have tried disabling and re-enabling in addition to uninstalling and re-installing the extension. 

Has anyone else experienced this?

Update: as noted below, this issue is unrelated to the chrome update. This issue is related to a Spotify update.

Solution below

**Comment by odiseo9999** [comment]
also me  , is not working , hope was for any actualization of chrome or security of window ,  i need the aplication is very good utility for me , please help.

**Comment by fabianramosg** [comment]
Yes, although firefox stopped working the same way, I don't know if it's because of chrome or spotify, but hopefully the developer can update this valuable tool.

**Comment by the-eduardo** [comment]
Same here :S

**Comment by TrickyPatrick** [comment]
Same, doesn't work anymore, please fix <3

**Comment by TrickyPatrick** [comment]
btw I don't think it's chrome issue but spotify issue. My chrome is still in 86 version but the same issue appears

**Comment by faf0** [comment]
I suspect that the following element is missing now in Spotify's DOM:
https://github.com/intOrfloat/spotitySpeedExtension/blob/cf65506c8382a3f7006f1ea401890eb20956ee80/content-script.js#L76

**Comment by faf0** [comment]
Yes, the fix is to replace line
https://github.com/intOrfloat/spotitySpeedExtension/blob/cf65506c8382a3f7006f1ea401890eb20956ee80/content-script.js#L76
by
```
				document.getElementsByClassName('volume-bar')[0].appendChild(input); /* make our input exist on page */
```

**Comment by odiseo9999** [comment]
i dont know how to change that , please send photos to know where tu put that ? please.

**Comment by faf0** [comment]
@odiseo9999 Should be fixed on this branch: https://github.com/intOrfloat/spotitySpeedExtension/pull/8
You can pull the branch to your file system, and then load the unpacked extension into Chrome.

To be clear, the issue doesn't have anything to do with a Chrome update. Spotify updated their code and the extension needs to be updated to append the speed changer bar at a different location in the DOM.

**Comment by TrickyPatrick** [comment]
Thank you bud, I will try this out ASAP

**Comment by odiseo9999** [comment]
i still really dont know how to do that?

**Comment by faf0** [comment]
@odiseo9999 , it's not easy until the extension is updated in the Chrome Web Store, unfortunately.

In the meantime, you may load the branch from the PR with the fix to your filesystem: https://github.com/faf0/spotitySpeedExtension/tree/faf0/fix-issue-7
Next, you may load this folder with manifest.json as an unpacked extension into Chrome:
- https://stackoverflow.com/a/24577660
- https://developer.chrome.com/docs/extensions/mv3/getstarted/#manifest

Also, see https://github.com/faf0/spotitySpeedExtension/tree/faf0/fix-issue-7#install-from-github

**Comment by odiseo9999** [comment]
please , make a actualization , i really need this amazing aplication for my job .

**Comment by faf0** [comment]
@odiseo9999 , I updated my previous comment, linking more precise instructions.

You may try the steps from https://github.com/faf0/spotitySpeedExtension/tree/faf0/fix-issue-7#install-from-github , but make sure you download the code from https://github.com/faf0/spotitySpeedExtension/tree/faf0/fix-issue-7

Hope that helps.

**Comment by odiseo9999** [comment]
please update on the store because i canot is really dificoult , to do , because in dont know programation . thanks .
please tell when wiill update .

**Comment by TrickyPatrick** [comment]
@odiseo9999 he is not the one who created this. He gently offered us a fix because the creator is inactive. So he can't update it on google chrome store.

**Comment by odiseo9999** [comment]
ok , tricky ,i really dont know how to do , than faf0 for try to help , is too much dificoult for me , cant do nothing.

**Comment by ryan6416** [comment]
@faf0 fix is working! THANKS!!! 🚀🏆🥇

![image](https://user-images.githubusercontent.com/12979345/111881165-8bceef00-896c-11eb-8956-4870e47e0766.png)

**Comment by odiseo9999** [comment]
It’s easy for you Ryan but we still without solution , I don’t know where to put that codes in chrome

**Comment by odiseo9999** [comment]
Any One can send the photos to know how to do Step by step please

**Comment by Grekkzio** [comment]
someone knows how to modify source code in opera gx? 
or in the same chrome?

**Comment by faf0** [comment]
As @TrickyPatrick pointed out, I'm not the maintainer of the extension on the Chrome Web Store, so I can't update it there.

Steps to install the extension with the fix manually:
1.  Download this ZIP: https://github.com/faf0/spotitySpeedExtension/archive/refs/heads/faf0/fix-issue-7.zip
2. Extract the ZIP
3. In Chrome, browse to chrome://extensions
4. Remove the currently installed "Spotify Playback Speed Access" extension
5. Enable "Developer mode" using the toggle in the top-right corner
6. Click on "Load unpacked" and select the previously unzipped folder that has the `manifest.json` file
7. Ensure the extension is enabled by checking the corresponding toggle
8. Reload https://open.spotify.com/
9. After changing the playback speed by entering the desired target speed percentage, hit the "Enter" key

**Comment by odiseo9999** [comment]
faf0 you save mi life , the fixed works   , thank you , brother thats the solution , you have here a new friend 
God bless you bro.

**Comment by odiseo9999** [comment]
i know you are helping bud this explication was amazing bro , thank you very much .

**Comment by fabianramosg** [comment]
@faf0 Many thanks bro!!

**Comment by AlexADittrich** [comment]
> btw I don't think it's chrome issue but spotify issue. My chrome is still in 86 version but the same issue appears

Thanks for referencing this

**Comment by AlexADittrich** [comment]
> @odiseo9999 Should be fixed on this branch: #8
> You can pull the branch to your file system, and then load the unpacked extension into Chrome.
> 
> To be clear, the issue doesn't have anything to do with a Chrome update. Spotify updated their code and the extension needs to be updated to append the speed changer bar at a different location in the DOM.

Thanks for noting that this is unrelated to the chrome update

**Comment by CharlieGBla** [comment]
This fix works!
if any of you are not as tech-savvy as I am not, here are some simple steps.

1. download the folder from this link
[https://github.com/faf0/spotitySpeedExtension/tree/faf0/fix-issue-7](url)
2. hit windows key + E to open file explorer and find the file in your downloads
3. select the folder and extract the files to anywhere on your computer, just make sure you remember where
4. go to [chrome://extensions/](url) in the chrome browser
5. hit load unpacked and select the extracted folder
6. that's it, if you did it correctly, it will show up in your extensions
I didn't know how to do any of this, so thank you to all the people above who did!

**Comment by wolvmarine** [comment]
> This fix works!
> if any of you are not as tech-savvy as I am not, here are some simple steps.
> 
> 1. download the folder from this link
>    [https://github.com/faf0/spotitySpeedExtension/tree/faf0/fix-issue-7](url)
> 2. hit windows key + E to open file explorer and find the file in your downloads
> 3. select the folder and extract the files to anywhere on your computer, just make sure you remember where
> 4. go to [chrome://extensions/](url) in the chrome browser
> 5. hit load unpacked and select the extracted folder
> 6. that's it, if you did it correctly, it will show up in your extensions
>    I didn't know how to do any of this, so thank you to all the people above who did!

That worked perfectly on Opera for me Thanks so much.  I would have never figured this out on my own.

**Comment by EarthCaste** [comment]
> As @TrickyPatrick pointed out, I'm not the maintainer of the extension on the Chrome Web Store, so I can't update it there.
> 
> Steps to install the extension with the fix manually:
> 
> 1. Download this ZIP: https://github.com/faf0/spotitySpeedExtension/archive/refs/heads/faf0/fix-issue-7.zip
> 2. Extract the ZIP
> 3. In Chrome, browse to chrome://extensions
> 4. Remove the currently installed "Spotify Playback Speed Access" extension
> 5. Enable "Developer mode" using the toggle in the top-right corner
> 6. Click on "Load unpacked" and select the previously unzipped folder that has the `manifest.json` file
> 7. Ensure the extension is enabled by checking the corresponding toggle
> 8. Reload https://open.spotify.com/
> 9. After changing the playback speed by entering the desired target speed percentage, hit the "Enter" key

Bless you faf0, you wonderful person.

**Comment by voidSphere** [comment]
If anyone else is having trouble setting up, I found another [extension](https://chrome.google.com/webstore/detail/global-speed/jpbjcnkcffbooppibceonlgknpkniiff) that also supports Spotify.

https://github.com/polywock/globalSpeed

**Comment by wolvmarine** [comment]
Nice to have options

**Comment by guckyslush** [comment]
thank you so much for this @faf0
